src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -148,20 +208,32 @@ export default App.extend({
       this.setState({ filters: _.clone(attributes) });
     });
   },
-  beforeStart() {
-    const filter = this.getState().getEntityFilter();
-    return Radio.request('entities', `fetch:${ this.getState().getType() }:collection`, { filter });
-  },
-  onStart(options, collection) {
-    this.collection = collection;
+  showMultiEditButtonView() {
+    this.getChildApp('filters').stop();
 
-    const collectionView = new ListView({
-      collection: this.collection,
+    const multiEditButtonView = this.showChildView('filters', new MultiEditButtonView({
       state: this.getState(),
-      viewComparator: this.getComparator(),
+      collection: this.collection,
+    }));
+
+    this.listenTo(multiEditButtonView, {
+      'click:cancel': this.onCancelMultiSelect,
+      'click:select': this.onClickMultiSelect,
     });
+  },
+  onCancelMultiSelect() {
+    this.getState().clearSelected();
+    this.collectionView.triggerMethod('select:none');
+  },
+  onClickMultiSelect() {
+    if (_.keys(this.getState().getSelected(this.collection)).length === this.collection.length) {
+      this.getState().clearSelected();
+      this.collectionView.triggerMethod('select:none');
+      return;
+    }
 
-    this.showChildView('list', collectionView);
+    this.getState().selectAll(this.collection);
+    this.collectionView.triggerMethod('select:all');
```

Comment:
I think these could move to triggers on the state itself at the end of `clearSelected` / `selectAll` which the collectionView can listen and react to.

Reply:
Yup, I had mulled over that solution as well. I like that organization better as well.